crates/ty_python_semantic/src/types.rs

```diff
@@ -4276,14 +4304,24 @@ impl<'db> Type<'db> {
                 .into()
             }
 
-            Type::GenericAlias(_) => {
+            Type::GenericAlias(alias) => {
+                let instantiated = Type::instance(db, ClassType::from(alias));
+
+                let parameters = if alias.origin(db).is_known(db, KnownClass::Tuple) {
+                    let spec = alias.specialization(db).tuple(db);
+                    let mut parameter =
+                        Parameter::positional_only(Some(Name::new_static("iterable")))
+                            .with_annotated_type(instantiated);
```

Comment:
With the `Tuple::resize` method that #18948 introduces, we can make this work for _any_ tuple, not just one with exactly the same size as the type being constructed.  I also have plans to have `try_iterator` return a tuple spec describing the return values of the iterator, which would then give us what we need for this to work for any iterable type.

Reply:
> With the `Tuple::resize` method that #18948 introduces, we can make this work for _any_ tuple, not just one with exactly the same size as the type being constructed.

Hmm, not sure I totally follow this. Can you give an example of something that should be covered in this PR but isn't? :-)